src/plugin/leaflet.canvas-markers.js

```diff
@@ -283,8 +283,8 @@ function layerFactory(L) {
 
             this._ctx.drawImage(
                 marker.canvas_img,
-                pointPos.x - marker.options.icon.options.iconAnchor[0]-(this._topLeftOffset?this._topLeftOffset.x:0),
-                pointPos.y - marker.options.icon.options.iconAnchor[1]-(this._topLeftOffset?this._topLeftOffset.y:0),
+                pointPos.x - marker.options.icon.options.iconAnchor[0]-this._map._mapPane._leaflet_pos.x-(this._topLeftOffset?this._topLeftOffset.x:0),
+                pointPos.y - marker.options.icon.options.iconAnchor[1]-this._map._mapPane._leaflet_pos.y-(this._topLeftOffset?this._topLeftOffset.y:0),
```

Comment:
Here we are deep into leaflet internals.
May be we could use [containerPointToLayerPoint](https://leafletjs.com/reference-1.4.0.html#map-containerpointtolayerpoint) instead?